pkg/rc/replication_controller.go

```diff
@@ -236,6 +242,12 @@ func (rc *replicationController) WatchDesires(quit <-chan struct{}) <-chan error
 		}
 
 		for rcFields := range rcChanges {
+			select {
+			case <-shouldInitialCheck:
+				rc.checkMissingArtifacts(rcFields)
+			case <-shouldPeriodicCheck:
+				rc.checkMissingArtifacts(rcFields)
+			}
```

Comment:
you should include a `default:` case here as well. The `default` case will fire if neither of the other channels are ready, which is desirable here because we might have to wait an hour for either of the channels to fire and we don't want to block that long